edx_event_bus_kafka/internal/consumer.py

```diff
@@ -322,17 +339,33 @@ def add_arguments(self, parser):
             required=True,
             help='Type of signal to emit from consumed messages.'
         )
+        parser.add_argument(
+            '-o', '--offset_time',
+            nargs=1,
+            required=False,
+            default=None,
+            help='The timestamp (in ISO format) that we would like to set the consumers to read from on startup.'
+                  'Overrides existing offsets.'
+        )
 
     def handle(self, *args, **options):
         if not KAFKA_CONSUMERS_ENABLED.is_enabled():
             logger.error("Kafka consumers not enabled, exiting.")
             return
         try:
             signal = OpenEdxPublicSignal.get_signal_by_type(options['signal'][0])
+            if options['offset_time'][0] is not None:
+                try:
+                    offset_timestamp = int(datetime.fromisoformat(options['offset_time'][0]).timestamp())
```

Comment:
It might be safest to do the parsing here, but pass it around as a datetime object -- then convert it to seconds or milliseconds from epoch only at the place where we'll use it. (Alternatively, a name like `offset_timestamp_s` to indicate that it's denominated in seconds.)